In [ ]:
import pandas as pd

Loading Dataset

In [ ]:
df_customers = pd.read_csv("/content/customers.csv")
df_orders = pd.read_csv("/content/orders.csv")
df_products = pd.read_csv("/content/products.csv")
df_products_orders = pd.read_csv("/content/products_orders.csv")

Understanding the Dataset Shape and Data

Based on the code result, it is expected product dataset to contain 20 rows and no missing value

In [ ]:
df_products.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20 entries, 0 to 19
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   product_id    20 non-null     object
 1   product_name  20 non-null     object
 2   price         20 non-null     int64 
 3   category      20 non-null     object
 4   description   20 non-null     object
dtypes: int64(1), object(4)
memory usage: 932.0+ bytes


Based on the code result below, there are several type of product inside product dataframe. However few of them have incorrect name

In [ ]:
df_products['product_name'].unique()

array(['Oak Chair', 'Dining Tab', 'Modern Sof', 'Queen Bed', 'Storage Ca',
       'Office Des', 'Wall Shelf', 'Patio Set', 'Coffee Tab', 'TV Stand',
       'Bookshelf', 'Bar Stool', 'Nightstand', 'Wardrobe', 'Bench',
       'Drawer', 'Side Table', 'Recliner', 'Console', 'Mirror'],
      dtype=object)

Replacing incorrect product name

In [ ]:
df_products['product_name'].replace(to_replace="Storage Ca", value="Storage Cabinet",inplace=True)
df_products["product_name"].replace(to_replace="Modern Sof",value="Modern Sofa", inplace=True)
df_products["product_name"].replace(to_replace="Office Des", value="Office Desk", inplace=True)
df_products["product_name"].replace(to_replace="Coffee Tab", value="Coffee Table", inplace=True)

/tmp/ipykernel_3068/882239683.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df_products['product_name'].replace(to_replace="Storage Ca", value="Storage Cabinet",inplace=True)
/tmp/ipykernel_3068/882239683.py:2: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when d

Checking value in category column in product dataset

In [ ]:
df_products['category'].unique()

array(['Sofa', 'Table', 'Bed', 'Shelf', 'Outdoor', 'Cabinet', 'Desk'],
      dtype=object)

Checking statistical value on numerical column in dataset

In [ ]:
df_products.describe()

,price
count,20.000000
mean,956.300000
std,530.113404
min,98.000000
25%,564.750000
50%,957.000000
75%,1447.500000
max,1675.000000


Understanding the Dataset Shape and Data

Based on the code result, it is expected that customer dataset to contain 200 rows and no missing value

In [ ]:
df_customers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 3 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   customer_id            200 non-null    object
 1   customer_name          200 non-null    object
 2   customer_phone_number  200 non-null    int64 
dtypes: int64(1), object(2)
memory usage: 4.8+ KB


Checking value in refund column in products orders dataset

In [ ]:
df_products_orders['refund'].unique()

array([0, 1])

Merging to dataframe products orders and products using inner join

In [ ]:
df_po_p = pd.merge(df_products_orders, df_products, on='product_id', how='inner')


Understanding the Dataset Shape and Data

Based on the code result, it is expected that new dataset contain 1000 rows and no missing value

In [ ]:
df_po_p.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   product_id      1000 non-null   object
 1   order_id        1000 non-null   object
 2   service         1000 non-null   int64 
 3   refund          1000 non-null   int64 
 4   starting_date   1000 non-null   object
 5   finishing_date  1000 non-null   object
 6   region          1000 non-null   object
 7   rating          1000 non-null   int64 
 8   product_name    1000 non-null   object
 9   price           1000 non-null   int64 
 10  category        1000 non-null   object
 11  description     1000 non-null   object
dtypes: int64(4), object(8)
memory usage: 93.9+ KB


Checking statistical value on numerical column in merged dataset

In [ ]:
df_po_p.describe()

,service,refund,rating,price
count,1000.000000,1000.000000,1000.000000,1000.00000
mean,0.371000,0.048000,3.016000,939.43700
std,0.483314,0.213873,1.417658,526.32563
min,0.000000,0.000000,1.000000,98.00000
25%,0.000000,0.000000,2.000000,477.00000
50%,0.000000,0.000000,3.000000,834.00000
75%,1.000000,0.000000,4.000000,1470.00000
max,1.000000,1.000000,5.000000,1675.00000


Change service and refund column data type to string

In [ ]:
df_po_p['service'] = df_po_p['service'].astype(str)
df_po_p['refund'] = df_po_p['refund'].astype(str)

Checking unique value on refund column

In [ ]:
df_po_p["refund"].unique()

array(['0', '1'], dtype=object)

replace "1" and "0" to "True" and "False" in service column

In [ ]:
df_po_p['service'].replace(to_replace="1",value="True", inplace=True)
df_po_p['service'].replace(to_replace="0",value="False", inplace=True)

/tmp/ipykernel_3068/2883125859.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df_po_p['service'].replace(to_replace="1",value="True", inplace=True)
/tmp/ipykernel_3068/2883125859.py:2: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value,

Checking unique value on service column

In [ ]:
df_po_p["service"].unique()

array(['False', 'True'], dtype=object)

replace "1" and "0" to "True" and "False" in refund column

In [ ]:
df_po_p['refund'].replace(to_replace="0",value="False", inplace=True)
df_po_p['refund'].replace(to_replace="1",value="True", inplace=True)

/tmp/ipykernel_3068/3693044786.py:1: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  df_po_p['refund'].replace(to_replace="0",value="False", inplace=True)
/tmp/ipykernel_3068/3693044786.py:2: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value,

Checking unique value on refund column

In [ ]:
df_po_p["refund"].unique()

array(['False', 'True'], dtype=object)

Understanding the Dataset Shape and Data

Based on the code result, it is expected that merged dataset contain 1000 rows and no missing value

In [ ]:
df_po_p.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   product_id      1000 non-null   object
 1   order_id        1000 non-null   object
 2   service         1000 non-null   object
 3   refund          1000 non-null   object
 4   starting_date   1000 non-null   object
 5   finishing_date  1000 non-null   object
 6   region          1000 non-null   object
 7   rating          1000 non-null   int64 
 8   product_name    1000 non-null   object
 9   price           1000 non-null   int64 
 10  category        1000 non-null   object
 11  description     1000 non-null   object
dtypes: int64(2), object(10)
memory usage: 93.9+ KB


Create a new **month** column in the merged dataset by extracting the month from the **starting_date** column.

In [ ]:
df_po_p['month'] = df_po_p['starting_date'].str[5:7]

Create a new **year** column in the merged dataset by extracting the year from the **starting_date** column.

In [ ]:
df_po_p['year'] = df_po_p['starting_date'].str[0:4]

Convert starting_date data type column into datetime

In [ ]:
df_po_p['starting_date'] = pd.to_datetime(df['starting_date'])

Convert finishing_date data type column into datetime

In [ ]:
df_po_p['finishing_date'] = pd.to_datetime(df['finishing_date'])

Subtract the **starting_date** column from the **finishing_date** column to calculate the number of days each delivery took.

In [ ]:
df_po_p['day_difference'] = df_po_p['finishing_date'] - df_po_p['starting_date']

Checking statistical value on numerical column in merged dataset

In [ ]:
df.describe()

,level_0,Unnamed: 0,index,starting_date,finishing_date,rating,price_x,day_difference,price_y
count,1000.000000,1000.000000,1000.000000,1000,1000,1000.000000,1000.00000,1000.000000,1000.00000
mean,499.500000,499.500000,499.500000,2026-06-30 21:36:00,2026-07-11 14:12:28.800000,3.016000,939.43700,10.692000,939.43700
min,0.000000,0.000000,0.000000,2026-01-01 00:00:00,2026-01-06 00:00:00,1.000000,98.00000,1.000000,98.00000
25%,249.750000,249.750000,249.750000,2026-03-27 06:00:00,2026-04-04 18:00:00,2.000000,477.00000,5.000000,477.00000
50%,499.500000,499.500000,499.500000,2026-07-05 12:00:00,2026-07-17 12:00:00,3.000000,834.00000,10.000000,834.00000
75%,749.250000,749.250000,749.250000,2026-10-03 00:00:00,2026-10-14 00:00:00,4.000000,1470.00000,16.000000,1470.00000
max,999.000000,999.000000,999.000000,2026-12-31 00:00:00,2027-01-19 00:00:00,5.000000,1675.00000,21.000000,1675.00000
std,288.819436,288.819436,288.819436,NaN,NaN,1.417658,526.32563,6.179589,526.32563


Few lines of code below convert day_difference column into string, take 2 first character and store it on day_difference column. Lastly convert day_difference column into integer

In [ ]:
df_po_p['day_difference'] = df_po_p['day_difference'].astype(str)

In [ ]:
df_po_p['day_difference'] = df_po_p['day_difference'].str[0:2]
df_po_p['day_difference'] = df_po_p['day_difference'].astype(int)

Understanding the Dataset Shape and Data

Based on the code result, it is expected that merged dataset contain 1000 rows and no missing value

In [ ]:
df_po_p.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   product_id      1000 non-null   object        
 1   order_id        1000 non-null   object        
 2   service         1000 non-null   object        
 3   refund          1000 non-null   object        
 4   starting_date   1000 non-null   datetime64[ns]
 5   finishing_date  1000 non-null   datetime64[ns]
 6   region          1000 non-null   object        
 7   rating          1000 non-null   int64         
 8   product_name    1000 non-null   object        
 9   price           1000 non-null   int64         
 10  category        1000 non-null   object        
 11  description     1000 non-null   object        
 12  month           1000 non-null   object        
 13  year            1000 non-null   object        
 14  day_difference  1000 non-null   int64         
dtypes: da

#EDA

Giving default index to our merged dataset, and store our merged dataset and the new index to the variable df

In [ ]:
df = df_po_p.reset_index()

In [ ]:
df.groupby(by="rating").agg(
    number_of_order = ("index","count")
)

,number_of_order
rating,
1,197
2,198
3,204
4,194
5,207


According to the data, there are 207 orders that give 5 rating, and 197 order that give 1 rating

In [ ]:
df.groupby(["product_name", "region"]).agg(
    rating = ("rating","mean"),
).reset_index().sort_values("rating").head(5)

,product_name,region,rating
59,Office Desk,West,1.916667
7,Bench,North,2.000000
43,Modern Sofa,South,2.000000
32,Drawer,North,2.125000
35,Mirror,Central,2.181818


According to the data, office desk (west showroom), bench (north showroom), modern sofa (south showroom), drawer (north showroom), mirorr (central showroom) have the lowest rating in the dataset

In [ ]:
df.groupby(["product_name", "region"]).agg(
    rating = ("rating","mean"),
).reset_index().sort_values("rating",ascending=False).head(3)

,product_name,region,rating
67,Queen Bed,North,4.250000
16,Coffee Table,East,4.200000
84,Storage Cabinet,West,4.166667


According to the data, queen bed (north showroom), coffe table (east showroom), and storage cabinet (west showroom) have the highest rating in the dataset

In [ ]:
df.groupby("region").agg(
    rating_on_avg = ("rating","mean"), max_rating = ("rating","max"), min_rating = ("rating","min")
)

,rating_on_avg,max_rating,min_rating
region,,,
Central,2.880829,5,1
East,3.151220,5,1
North,2.967213,5,1
South,3.095238,5,1
West,2.971292,5,1


According to the data, Central showroom has the lowest rating among all showroom that that this furniture company have, meanwhile the highest rating hold by west showroom

In [ ]:
df[df["region"] == "Central"].groupby("product_name").agg(
    number_of_order = ("index","count")
).reset_index().sort_values("number_of_order", ascending=False)

,product_name,number_of_order
13,Queen Bed,14
6,Drawer,14
0,Bar Stool,13
10,Oak Chair,13
12,Patio Set,13
4,Console,13
15,Side Table,12
7,Mirror,11
5,Dining Tab,11
8,Modern Sofa,10


According to the data, the most in-demand items in the **Central** showroom are the **Queen Bed, Drawer, Barstool, Oak Chair, Patio Set, and Console**. The least purchased product is the **Storage Cabinet**.

In [ ]:
df[df["region"] == "East"].groupby("product_name").agg(
    number_of_order = ("index","count")
).reset_index().sort_values("number_of_order", ascending=False)

,product_name,number_of_order
0,Bar Stool,17
17,TV Stand,14
18,Wall Shelf,14
5,Dining Tab,14
11,Office Desk,13
16,Storage Cabinet,12
19,Wardrobe,12
4,Console,11
14,Recliner,11
10,Oak Chair,10


According to the data, the most in-demand items in the **East** showroom are the **Bar stool, TV stand, Wall Shelf, Dining Tab**. The least purchased product is the **Coffee Table**.

In [ ]:
df[df["region"] == "North"].groupby("product_name").agg(
    number_of_order = ("index","count")
).reset_index().sort_values("number_of_order", ascending=False)

,product_name,number_of_order
3,Coffee Table,15
11,Office Desk,15
9,Nightstand,14
10,Oak Chair,12
17,TV Stand,10
19,Wardrobe,10
2,Bookshelf,9
15,Side Table,9
18,Wall Shelf,9
5,Dining Tab,9


According to the data, the most in-demand items in the **North** showroom are the **Coffe Table, Office Desk, Nightstand**. The least purchased product is the **Mirror**.

In [ ]:
df[df["region"] == "South"].groupby("product_name").agg(
    number_of_order = ("index","count")
).reset_index().sort_values("number_of_order", ascending=False)

,product_name,number_of_order
2,Bookshelf,17
3,Coffee Table,15
0,Bar Stool,14
9,Nightstand,14
18,Wall Shelf,13
14,Recliner,12
11,Office Desk,12
7,Mirror,12
16,Storage Cabinet,12
1,Bench,11


According to the data, the most in-demand items in the **South** showroom are the **Bookshelf**. The least purchased product is the **Dining Tab**.

In [ ]:
df[df["region"] == "West"].groupby("product_name").agg(
    number_of_order = ("index","count")
).reset_index().sort_values("number_of_order", ascending=False)

,product_name,number_of_order
17,TV Stand,18
10,Oak Chair,15
13,Queen Bed,15
2,Bookshelf,13
19,Wardrobe,13
15,Side Table,12
11,Office Desk,12
6,Drawer,12
12,Patio Set,12
16,Storage Cabinet,12


According to the data, the most in-demand items in the **West** showroom are the **TV Stand**. The least purchased product is the **Console**.

In [ ]:
df.groupby(["region"]).agg(number_of_order = ("index","count"))

,number_of_order
region,
Central,193
East,205
North,183
South,210
West,209


According to the data, the **South** showroom has the highest number of orders in this dataset, while the **North** showroom has the lowest number of orders.

In [ ]:
df.groupby(["month","region"]).agg(number_of_order = ("index","count"))

number_of_order
month region                  
01    Central               17
      East                  25
      North                 16
      South                 21
      West                  19
02    Central               19
      East                  14
      North                 14
      South                 19
      West                  17
03    Central               13
      East                  14
      North                 16
      South                 20
      West                  22
04    Central               10
      East                  13
      North                  5
      South                 22
      West                  23
05    Central               19
      East                  17
      North                 21
      South                 11
      West                  12
06    Central               17
      East                  11
      North                 15
      South                 15
      West                  16
07    Central               11
      East                  17
      North                 12
      South                 12
      West                  16
08    Central               20
      East                  28
      North                 17
      South                 24
      West                  15
09    Central               15
      East                  12
      North                 14
      South                  9
      West                  23
10    Central               19
      East                  19
      North                 18
      South                 20
      West                  21
11    Central               15
      East                  15
      North                 20
      South                 22
      West                  14
12    Central               18
      East                  20
      North                 15
      South                 15
      West                  11

According to the data:
*   On the first month east showroom handle the highest number of order, and the least is north showroom
*   On the second month South and Central showroom handle the highest number of order, and the least are east and north showroom
*   On the third month West showroom handle the highest number of order, and the least is Central showroom
*   On the forth month West showroom handle the highest number of order, and the least is North showroom
*   On the Fifth month North showroom handle the highest number of order, and the least is South showroom
*   On the Sixth month Central showroom handle the highest number of order, and the least is East showroom
*   On the seventh month East showroom handle the highest number of order, and the least are North and South showroom
*   On the eighth month East showroom handle the highest number of order, and the least is West showroom
*   On the ninth month West showroom handle the highest number of order, and the least is South showroom
*   On the tenth month West showroom handle the highest number of order, and the least is North showroom
*   On the eleventh month South showroom handle the highest number of order, and the least is West showroom
*   On the ninth month East showroom handle the highest number of order, and the least is West showroom








In [ ]:
df.groupby(["month"]).agg(number_of_order = ("index","count"))

,number_of_order
month,
01,98
02,83
03,85
04,73
05,80
06,74
07,68
08,104
09,73


According to the data, the highest number of order is on August, followed by january, and october. The lowest number of order is on July

In [ ]:
df.groupby(["region","refund"]).agg(number_of_order = ("index","count"))

number_of_order
region  refund                 
Central False               181
        True                 12
East    False               195
        True                 10
North   False               180
        True                  3
South   False               200
        True                 10
West    False               196
        True                 13

According to the data, most products from all of our showrooms do not require refunds to customers. However, the **Central** showroom has the highest number of refunds compared to the other showrooms.

In [ ]:
df[df["refund"] == "True"].groupby(["region","product_name"]).agg(number_of_order = ("index","count"))

number_of_order
region  product_name                    
Central Bar Stool                      1
        Bookshelf                      1
        Coffee Table                   3
        Mirror                         1
        Nightstand                     1
        Oak Chair                      2
        Office Desk                    1
        Queen Bed                      1
        Side Table                     1
East    Bar Stool                      2
        Bench                          1
        Console                        1
        Dining Tab                     2
        Drawer                         1
        Modern Sofa                    1
        Recliner                       1
        Wardrobe                       1
North   Office Desk                    1
        Recliner                       1
        Wardrobe                       1
South   Bookshelf                      3
        Coffee Table                   1
        Mirror                         1
        Modern Sofa                    1
        Oak Chair                      1
        Storage Cabinet                1
        TV Stand                       1
        Wall Shelf                     1
West    Bar Stool                      1
        Bookshelf                      2
        Dining Tab                     1
        Drawer                         1
        Mirror                         2
        Storage Cabinet                1
        TV Stand                       3
        Wardrobe                       2

According to the data, most products from all of our showrooms do not require refunds to customers.

In [ ]:
df_service = df[df["service"] == "True"]
df_service[df_service["region"] == "Central"].groupby(["product_name"]).agg(number_of_order = ("index","count")).sort_values("number_of_order", ascending=False).head(3)

,number_of_order
product_name,
Console,9
Patio Set,8
Modern Sofa,6


According to the data, top 3 product that need to be service that come from central showroom are console, patio set, and moderns sofa

In [ ]:
df_service[df_service["region"] == "East"].groupby(["product_name"]).agg(number_of_order = ("index","count")).sort_values("number_of_order", ascending=False).head(3)

,number_of_order
product_name,
Storage Cabinet,7
TV Stand,7
Bar Stool,6


According to the data, top 3 product that need to be service that come from East showroom are Storage Cabinet, TV Stand, Bar Stool

In [ ]:
df_service[df_service["region"] == "West"].groupby(["product_name"]).agg(number_of_order = ("index","count")).sort_values("number_of_order", ascending=False).head(3)

,number_of_order
product_name,
Queen Bed,6
Wall Shelf,6
Bookshelf,5


According to the data, top 3 product that need to be service that come from west showroom are Queen Bed, Wall Shelf, BookShelf

In [ ]:
df_service[df_service["region"] == "South"].groupby(["product_name"]).agg(number_of_order = ("index","count")).sort_values("number_of_order", ascending=False).head(3)

,number_of_order
product_name,
Wardrobe,7
Office Desk,7
Recliner,6


According to the data, top 3 product that need to be service that come from South showroom are Wardrobe, Office Desk, and Recliner

In [ ]:
df_service[df_service["region"] == "North"].groupby(["product_name"]).agg(number_of_order = ("index","count")).sort_values("number_of_order", ascending=False).head(3)

,number_of_order
product_name,
Coffee Table,6
Oak Chair,5
Nightstand,5


According to the data, top 3 product that need to be service that come from North showroom are Coffee Table, Oak Chair, Nightstand

In [ ]:
df[df["region"] == "Central"].groupby("product_name").agg(preparation = ("day_difference","mean"), rating_on_avg = ("rating","mean")).sort_values("preparation",ascending=False).head(3)

,preparation,rating_on_avg
product_name,,
Bench,15.285714,2.285714
Bar Stool,13.769231,3.307692
Patio Set,13.076923,3.461538


According to the data, top 3 product take the longest time to produce in Central Showroom are Bench, Bar Stool, Patio Set

In [ ]:
df[df["region"] == "East"].groupby("product_name").agg(preparation = ("day_difference","mean"), rating_on_avg = ("rating","mean")).sort_values("preparation",ascending=False).head(3)

,preparation,rating_on_avg
product_name,,
Nightstand,14.250000,2.750000
Wardrobe,13.083333,3.000000
Patio Set,12.666667,3.777778


According to the data, top 3 product take the longest time to produce in East Showroom are Nightstand, Wardrobe, Patio Set

In [ ]:
df[df["region"] == "West"].groupby("product_name").agg(preparation = ("day_difference","mean"), rating_on_avg = ("rating","mean")).sort_values("preparation",ascending=False).head(3)

,preparation,rating_on_avg
product_name,,
Dining Tab,15.400000,2.500000
Wall Shelf,14.714286,3.571429
Drawer,14.250000,3.000000


According to the data, top 3 product take the longest time to produce in West Showroom are Dining Tab, Wall Shelf, Drawer

In [ ]:
df[df["region"] == "North"].groupby("product_name").agg(preparation = ("day_difference","mean"), rating_on_avg = ("rating","mean")).sort_values("preparation",ascending=False).head(3)

,preparation,rating_on_avg
product_name,,
Patio Set,15.714286,2.571429
Bookshelf,15.666667,3.000000
Drawer,14.375000,2.125000


According to the data, top 3 product take the longest time to produce in North Showroom are Patio Set, Bookshelf, and Drawer

In [ ]:
df[df["region"] == "South"].groupby("product_name").agg(preparation = ("day_difference","mean"), rating_on_avg = ("rating","mean")).sort_values("preparation",ascending=False).head(3)

,preparation,rating_on_avg
product_name,,
Coffee Table,12.133333,3.533333
Office Desk,11.583333,3.083333
Console,11.333333,3.111111


According to the data, top 3 product take the longest time to produce in South Showroom are Coffee Table, Office Desk, and Console

In [ ]:
df.head()

,index,product_id,order_id,service,refund,starting_date,finishing_date,region,rating,product_name,price,category,description,month,year,day_difference
0,0,P0004,O0001,False,False,2026-04-17,2026-04-19,West,4,Queen Bed,1470,Bed,Synthetic furniture,04,2026,2
1,1,P0018,O0002,False,False,2026-11-18,2026-12-06,North,4,Recliner,602,Shelf,Synthetic furniture,11,2026,18
2,2,P0010,O0003,False,False,2026-08-21,2026-09-01,East,2,TV Stand,477,Shelf,Synthetic furniture,08,2026,11
3,3,P0008,O0004,True,False,2026-11-30,2026-12-04,East,1,Patio Set,661,Shelf,Synthetic furniture,11,2026,4
4,4,P0013,O0005,True,False,2026-12-13,2026-12-29,Central,2,Nightstand,725,Desk,Synthetic furniture,12,2026,16


In [ ]:
df.to_csv("products_orders_finish.csv")